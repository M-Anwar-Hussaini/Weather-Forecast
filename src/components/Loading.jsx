import Load from 'react-loading';

export default function Loading() {
  return (
    <div className="d-flex flex-column align-items-center justify-content-center gap3 p-3 bgcolor-dark">
      <Load type="spokes" />
      <span>Please wait...</span>
    </div>
  );
}
