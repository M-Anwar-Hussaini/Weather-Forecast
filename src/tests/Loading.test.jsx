import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Loading from '../components/Loading';

describe('Apply tests for Loading component', () => {
  test('Take snapshot from Loading', () => {
    const loading = render(<Loading />);
    expect(loading).toMatchSnapshot();
  });

  test('Take snapshot from Loading', () => {
    const loading = render(<Loading />);
    expect(loading).toMatchSnapshot();
  });

  test('Loading has an element with the text of please wait', () => {
    render(<Loading />);
    const text = screen.getByText(/Please wait.../i);
    expect(text).toBeInTheDocument();
  });
});
