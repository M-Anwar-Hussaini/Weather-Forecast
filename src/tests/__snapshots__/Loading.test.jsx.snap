// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apply tests for Loading component Take snapshot from Loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex flex-column align-items-center justify-content-center gap3 p-3 bgcolor-dark"
      >
        <div
          style="fill: #fff; height: 64px; width: 64px;"
        >
          <svg
            id="loading"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(0 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(45 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.125s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(90 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.25s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(135 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.375s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(180 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.5s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(225 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.675s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(270 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.75s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(315 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.875s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            

          </svg>
          

        </div>
        <span>
          Please wait...
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex flex-column align-items-center justify-content-center gap3 p-3 bgcolor-dark"
    >
      <div
        style="fill: #fff; height: 64px; width: 64px;"
      >
        <svg
          id="loading"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(0 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(45 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.125s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(90 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.25s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(135 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.375s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(180 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.5s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(225 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.675s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(270 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.75s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(315 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.875s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          

        </svg>
        

      </div>
      <span>
        Please wait...
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Apply tests for Loading component Take snapshot from Loading 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex flex-column align-items-center justify-content-center gap3 p-3 bgcolor-dark"
      >
        <div
          style="fill: #fff; height: 64px; width: 64px;"
        >
          <svg
            id="loading"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(0 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(45 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.125s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(90 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.25s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(135 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.375s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(180 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.5s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(225 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.675s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(270 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.75s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            
  
            <path
              d="M14 0 H18 V8 H14 z"
              opacity=".1"
              transform="rotate(315 16 16)"
            >
              
    
              <animate
                attributeName="opacity"
                begin="0.875s"
                dur="1s"
                from="1"
                repeatCount="indefinite"
                to=".1"
              />
              
  
            </path>
            

          </svg>
          

        </div>
        <span>
          Please wait...
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex flex-column align-items-center justify-content-center gap3 p-3 bgcolor-dark"
    >
      <div
        style="fill: #fff; height: 64px; width: 64px;"
      >
        <svg
          id="loading"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(0 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(45 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.125s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(90 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.25s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(135 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.375s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(180 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.5s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(225 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.675s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(270 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.75s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          
  
          <path
            d="M14 0 H18 V8 H14 z"
            opacity=".1"
            transform="rotate(315 16 16)"
          >
            
    
            <animate
              attributeName="opacity"
              begin="0.875s"
              dur="1s"
              from="1"
              repeatCount="indefinite"
              to=".1"
            />
            
  
          </path>
          

        </svg>
        

      </div>
      <span>
        Please wait...
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
